cloudfoundry/resource_cf_app.go

```diff
@@ -713,6 +698,12 @@ func resourceAppUpdate(d *schema.ResourceData, meta interface{}) error {
 	//       service bindings are updates?)
 	app.DockerImage = getChangedValueString("docker_image", &update, d)
 	app.DockerCredentials = getChangedValueMap("docker_credentials", &update, d)
+	if app.DockerImage != nil && *app.DockerImage == "" {
+		app.DockerImage = nil
+	}
+	if app.DockerImage != nil && d.HasChange("docker_image") {
+		app.Ports = nil
```

Comment:
Why nil out `app.Ports` when `d.HasChange("docker_image")`?

Reply:
+1, I have the same question

Reply:
@janosbinder could this be related to symptoms reported by @lixilin2301 in https://github.com/mevansam/terraform-provider-cf/issues/179 ?